common/p_spec.cpp

```diff
@@ -82,6 +82,97 @@ std::list<movingsector_t>::iterator P_FindMovingSector(sector_t *sector)
 	return movingsectors.end();
 }
 
+int P_FindLineFromLineTag(const line_t* line, int start)
+{
+	start = start >= 0 ? lines[start].nextid
+	                   : lines[(unsigned)line->id % (unsigned)numlines].firstid;
+	while (start >= 0 && lines[start].id != line->id)
+		start = lines[start].nextid;
+	return start;
+}
+
+/*
+ * P_IsUnderDamage
+ *
+ * killough 9/9/98:
+ *
+ * Returns nonzero if the object is under damage based on
+ * their current position.
+ */
+int P_IsUnderDamage(AActor* actor)
+{
+	const struct msecnode_s* seclist;
+	const DCeiling* cr; // Crushing ceiling
+	int dir = 0;
+	for (seclist = actor->touching_sectorlist; seclist; seclist = seclist->m_tnext)
+		if ((cr = (DCeiling*)seclist->m_sector->ceilingdata) &&
+		    cr->m_Status == 2) // Down
+			cr->m_Crush ? dir = 1 : dir = 0;
+	return dir;
+}
+
+/*
+* 
+* P_IsFriendlyThing
+* @brief Helper function to determine if a particular thing is of friendly origin.
+* 
+* @param actor Source actor
+* @param friendshiptest Thing to test friendliness
+*/
+bool P_IsFriendlyThing(AActor* actor, AActor* friendshiptest)
+{
+	if (friendshiptest->flags & MF_FRIEND)
+	{
+		if (sv_gametype == (GM_COOP || GM_HORDE))
+		{
+			return true;
+		}
+		else if (actor->player && friendshiptest->target && friendshiptest->target->player &&
+		    actor->player->userinfo.team == friendshiptest->target->player->userinfo.team)
+		{
+			return true;
+		}
+		else
+		{
+			return false;
+		}
+	}
+	else
+	{
+		return false;
+	}
+}
+
+/*
+ *
+ * P_IsTeamMate
+ * @brief Helper function to determine if a particular thing is a teammate.
+ *
+ * @param actor Source player actor
+ * @param player Player actor to test same teamness
+ * @returns A boolean to determine if the player is on the shooter's team.
+ */
+bool P_IsTeamMate(AActor* actor, AActor* player)
```

Comment:
Seems like duplicated functionality with `P_AreTeammates`.  See if you can consolidate this down to one function, whichever one makes more sense.